universal-application-tool-0.0.1/app/controllers/admin/TrustedIntermediaryManagementController.java

```diff
@@ -20,7 +22,18 @@ public TrustedIntermediaryManagementController(
   }
 
   @Secure(authorizers = Authorizers.Labels.UAT_ADMIN)
-  public Result index() {
-    return ok(listView.render(userRepository.listTrustedIntermediaryGroups()));
+  public Result index(Http.Request request) {
+    return ok(listView.render(userRepository.listTrustedIntermediaryGroups(), request));
+  }
+
+  @Secure(authorizers = Authorizers.Labels.UAT_ADMIN)
+  public Result newGroup(Http.Request request) {
+    String[] name = request.body().asFormUrlEncoded().get("name");
+    String[] description = request.body().asFormUrlEncoded().get("description");
+    Preconditions.checkState(name.length == 1, "Need exactly one `name` form entry.");
+    Preconditions.checkState(description.length == 1, "Need exactly one `name` form entry.");
```

Comment:
It looks like `Preconditions.checkState` throws an exception if the requirements aren't met, which I don't think we want to do here. I think we want to return a friendly validation error message and re-render the form instead.

The standard way we've been handling forms/POSTs is via `Form` classes. See [BlockForm](https://github.com/seattle-uat/civiform/blob/main/universal-application-tool-0.0.1/app/forms/BlockForm.java) for a simple 2-field example. And see how it's used in [AdminProgramBlocksController](https://github.com/seattle-uat/civiform/blob/b7925499481906412758407e9b661cebe1f02bfc/universal-application-tool-0.0.1/app/controllers/admin/AdminProgramBlocksController.java#L99).

I think creating a `TrustedIntermediaryGroupForm` might help streamline the code to re-render the form with the values the user (admin) originally provided. Let me know if I can help.